components/eam/src/chemistry/mozart/chemistry.F90

```diff
@@ -1555,10 +1559,15 @@ subroutine chem_timestep_tend( state, ptend, cam_in, cam_out, dt, pbuf,  fh2o, f
              else
                 Diff(:ncol,1) = (ftem_layers(:ncol,1) - gas_ac_2D(:ncol))/dt
              end if
+             if (history_chemdyg_summary .and. trim(solsym(n)) == 'O3' ) then
+             call outfld(trim(solsym(n))//'_2DTDO', Diff(:ncol,1), pcols, lchnk )
+             end if
+             if (history_gaschmbudget_2D ) then
              call outfld(trim(solsym(n))//'_2DTDO', Diff(:ncol,1), pcols, lchnk )
+             end if
```

Comment:
It is possible that both `history_chemdyg_summary` and `history_gaschmbudget_2D` are true and it is redundant to write out same variables twice and might be problematic. It would be cleaner to combine these two if-statements into one:
```
if (history_gaschmbudget_2D .or. (history_chemdyg_summary .and. trim(solsym(n)) == 'O3') then
    call outfld(trim(solsym(n))//'_2DTDO', Diff(:ncol,1), pcols, lchnk )
end if
```